### DISCUSSION TITLE
How to not ignore the first value?

### DISCUSSION THREAD

**Discussion by davhdavh** [open-telemetry/opentelemetry-dotnet#5842] [Q&A]
As it is now 
```
   private readonly Counter<long> _myCounter= meter.CreateCounter<long>("usage.dcs.duration_milliseconds", "ms", "Milliseconds used running xxx");
...
_myCounter.Add((int)elapsedTime.TotalMilliseconds, tag1, tag2);
```

The first usage of Add for every unique tag combination is "ignored".

The problem is that promql's `increase` function only reports on INCREASES, and thus the very first reported data-point is also the "base value".

How do I force metrics to start from 0?
Is there a secret setting in dotnet, or in the counter or in otel collector or prometheus or grafana for this problem?
Or metrics only works if you know in advance what your dimension values are?

EDIT:
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/data-model.md#resets-and-gaps
has a tiny section that says:
```
When StartTimeUnixNano is less than TimeUnixNano, a new unbroken sequence of observations begins with a "true" reset at a known start time. The zero value is implicit, it is not necessary to record the starting point.
```
So how to get it to emit this properly?

**Comment by TimothyMothra** [comment]
Hi @davhdavh,
No metric points should be dropped.
Can you help us by creating a sample app using the Console Exporter that repros this issue? 
If you can provide that we can take a look.

**Reply by davhdavh** [reply]
No, they are not dropped... They are just ignored due to how the `increase` function works.
`sum by(tag1, tag2) (increase(mycounter{}[$__interval]))` will always ignore the very first usage for each tag set for each labelset.
ie, if the first request for tag1=x and tag2=y takes 200ms, and 2nd takes 10ms, the reported usage is 10ms.
e.g. I have another counter where I also store the usage in db for accounting. DB says 28427 items used yesterday, `increase` of the metrics counter that report the same data says 19991 used, and they can all be tracked down to the first usage for each set of tags is ignored.

**Reply by cijothomas** [reply]
Looks like a prometheus query issue? It may be a good question to Prometheus forum.

(Also, OTel .NET does not do resets for Cumulative ever)
